src/Roslyn/Conventional.Roslyn/ThisCodebase.cs

```diff
@@ -21,13 +21,13 @@ public static IEnumerable<ConventionResult> MustConformTo(ISolutionDiagnosticAna
 
             var solution = workspace.OpenSolutionAsync(KnownPaths.FullPathToSolution).Result;
 
-            foreach(var diagnostic in workspace.Diagnostics)
+            foreach (var diagnostic in workspace.Diagnostics)
             {
                 Trace.WriteLine(diagnostic.Message);
             }
 
             return Conformist.EnforceConformance(
-                convention.IsSatisfiedBy(solution));
+                convention.IsSatisfiedBy(solution, knownOffenders));
```

Comment:
I think I'd prefer to keep the known offender capability separate from the convention abstraction similar to the non-roslyn implementation. Could you pull the filtering here into `ThisCodebase` instead please?

Reply:
In retrospect, this is rather obvious. Can definitely make this change. 👍